geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateJndiBindingCommand.java

```diff
@@ -0,0 +1,260 @@
+          help = JDBC_DRIVER_CLASS__HELP) String jdbcDriver,
+      @CliOption(key = JNDI_NAME, mandatory = true, help = JNDI_NAME__HELP) String jndiName,
+      @CliOption(key = LOGIN_TIMEOUT_SECONDS,
+          help = LOGIN_TIMEOUT_SECONDS__HELP) Integer loginTimeout,
+      @CliOption(key = MANAGED_CONN_FACTORY_CLASS,
+          help = MANAGED_CONN_FACTORY_CLASS__HELP) String managedConnFactory,
+      @CliOption(key = MAX_POOL_SIZE, help = MAX_POOL_SIZE__HELP) Integer maxPoolSize,
+      @CliOption(key = PASSWORD, help = PASSWORD__HELP) String password,
+      @CliOption(key = TRANSACTION_TYPE, help = TRANSACTION_TYPE__HELP) String transactionType,
+      @CliOption(key = TYPE, mandatory = true,
+          help = TYPE__HELP) JndiBindingConfiguration.DATASOURCE_TYPE type,
+      @CliOption(key = USERNAME, help = USERNAME__HELP) String username,
+      @CliOption(key = XA_DATASOURCE_CLASS, help = XA_DATASOURCE_CLASS__HELP) String xaDataSource,
+      @CliOption(key = CliStrings.IFNOTEXISTS, help = IFNOTEXISTS__HELP,
+          specifiedDefaultValue = "true", unspecifiedDefaultValue = "false") boolean ifNotExists,
+      @CliOption(key = DS_CONFIG_PROPERTIES, optionContext = "splittingRegex=,(?![^{]*\\})",
+          help = DS_CONFIG_PROPERTIES_HELP) ConfigProperty[] dsConfigProperties)
+      throws IOException, SAXException, ParserConfigurationException, TransformerException {
+
+    boolean alreadyExists = isBindingAlreadyExists(jndiName);
+    if (alreadyExists && ifNotExists)
+      return ResultBuilder.createInfoResult(CliStrings.format(
+          "Skipping creation of Jndi Binding with jndi-name \"{0}\" as its already exists.",
+          jndiName));
+    else if (alreadyExists)
+      return ResultBuilder.createUserErrorResult(
+          CliStrings.format("JNDI Binding with jndi-name \"{0}\" already exists.", jndiName));
+
+    JndiBindingConfiguration configuration = new JndiBindingConfiguration();
+    configuration.setBlockingTimeout(blockingTimeout);
+    configuration.setConnectionPoolDatasource(connectionPooledDatasource);
+    configuration.setConnectionUrl(connectionUrl);
+    configuration.setIdleTimeout(idleTimeout);
+    configuration.setInitPoolSize(initPoolSize);
+    configuration.setJdbcDriver(jdbcDriver);
+    configuration.setJndiName(jndiName);
+    configuration.setLoginTimeout(loginTimeout);
+    configuration.setManagedConnFactory(managedConnFactory);
+    configuration.setMaxPoolSize(maxPoolSize);
+    configuration.setPassword(password);
+    configuration.setTransactionType(transactionType);
+    configuration.setType(type);
+    configuration.setUsername(username);
+    configuration.setXaDatasource(xaDataSource);
+    if (dsConfigProperties != null && dsConfigProperties.length > 0)
+      configuration.setDsConfigurations(Arrays.asList(dsConfigProperties));
+
+    ClusterConfigurationService service = getSharedConfiguration();
+
+    if (service == null) {
```

Comment:
I think even though cc is not available, we should still allow them to execute the function to create the jndi binding on the servers, we just don't update cluster config. that's what all the other commands are doing.

Reply:
Creation of jndi bindings does not need for servers to be up and we do want to users to use CC, so failing when CC unavailable is certainly encourages our users to use it. Also if users are not using CC then they may be more inclined to use cache.xml?

I see `AlterAsyncEventQueueCommand` errors if CC is not running.